integration/unplugin/src/index.ts

```diff
@@ -91,29 +73,37 @@ function resolveAbsolutePath(rootDir: string, id: string) {
 }
 
 const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
-  if (options.dts && options.js) {
-    throw new Error("Can't have both `dts` and `js` be set to `true`.");
-  }
-
-  if (options.typecheck && options.js) {
-    throw new Error("Can't have both `typecheck` and `js` be set to `true`.");
-  }
-
-  const transpileToJS = options.js ?? false;
-  // When Civet's js option is better, we could consider a different default:
-  //const transpileToJS = options.js ?? !(options.dts || options.typecheck);
-  const outExt = options.outputExtension ?? (transpileToJS ? '.jsx' : '.tsx');
+  const transformTS = options.emitDeclaration || options.typecheck;
+  const outExt =
+    options.outputExtension ?? (options.ts === 'preserve' ? '.tsx' : '.jsx');
 
   let fsMap: Map<string, string> = new Map();
   const sourceMaps = new Map<string, SourceMap>();
   let compilerOptions: any;
   let rootDir = process.cwd();
 
+  const tsPromise =
+    transformTS || options.ts === 'tsc'
+      ? import('typescript').then(m => m.default)
+      : null;
+  const getTS = () => tsPromise!;
```

Comment:
Not sure what `getTS` is for... why not just use `tsPromise!` directly? In fact, you do that at least once below already. 🙂

Reply:
Just so we don't have to use the non-null assertion everywhere. but now that i think of it, its probably better to remove it so we know that we are accessing something that might be null :)